specs/_features/maxeb_increase/capella.py

```diff
@@ -3540,24 +3559,38 @@ def has_eth1_withdrawal_credential(validator: Validator) -> bool:
     return validator.withdrawal_credentials[:1] == ETH1_ADDRESS_WITHDRAWAL_PREFIX
 
 
+def has_compounding_withdrawal_credential(validator: Validator) -> bool:
+    """
+    Check if ``validator`` has an 0x02 prefixed "compounding" withdrawal credential.
+    """
+    return validator.withdrawal_credentials[:1] == COMPOUNDING_WITHDRAWAL_PREFIX
+
+
 def is_fully_withdrawable_validator(validator: Validator, balance: Gwei, epoch: Epoch) -> bool:
     """
     Check if ``validator`` is fully withdrawable.
     """
     return (
-        has_eth1_withdrawal_credential(validator)
+        (has_eth1_withdrawal_credential(validator) or has_compounding_withdrawal_credential(validator))
         and validator.withdrawable_epoch <= epoch
         and balance > 0
     )
 
+def get_validator_excess_balance(validator: Validator, balance: Gwei) -> Gwei:
+    """
+    Get excess balance for paritial withdrawals for ``validator``.
+    """
+    if has_compounding_withdrawal_credential(validator) and balance > MAX_EFFECTIVE_BALANCE:
+        return balance - MAX_EFFECTIVE_BALANCE
+    elif has_eth1_withdrawal_credential(validator) and balance > MIN_ACTIVATION_BALANCE:
+        return balance - MIN_ACTIVATION_BALANCE
+    return 0
```

Comment:
Consider `return Gwei(0)` instead of `return 0`. No exactly sure what the rule is. There may be a type linter/checker—may be worth asking Hsiao Wei :)